tests/Umbraco.Tests.UnitTests/Umbraco.Core/PropertyEditors/RichTextPropertyIndexValueFactoryTests.cs

```diff
@@ -0,0 +1,84 @@
+    [TestCase("<p>Text with <blockquote>quoted text</blockquote></p>", "Text with quoted text")]
+    [TestCase("<p>Text with <ul><li>list item 1</li><li>list item 2</li></ul></p>",
+        "Text with list item 1list item 2 ")]
+    [TestCase("<p>Text with <ol><li>ordered item 1</li><li>ordered item 2</li></ol></p>",
+        "Text with ordered item 1ordered item 2")]
+    [TestCase("<p>Text with <div class=\"class-name\">div content</div></p>", "Text with div content")]
+    [TestCase("<p>Text with <span class=\"class-name\">span content</span></p>", "Text with span content")]
+    [TestCase("<p>Text with <strong>bold</strong> and <em>italic</em> content</p>",
+        "Text with bold and italic content")]
+    [TestCase("<p>Text with <a href=\"https://example.com\" target=\"_blank\">external link</a></p>",
+        "Text with external link")]
+    [TestCase("<p>John Smith<br class=\"test\">Company ABC<br>London</p>", "John Smith Company ABC London")]
+    [TestCase("<p>John Smith<br \r\n />Company ABC<br>London</p>", "John Smith Company ABC London")]
+    public void Can_Create_Index_Values_From_RichText_Property(string testContent, string expected)
+    {
+        var propertyEditorCollection = new PropertyEditorCollection(new DataEditorCollection(() => null));
+        var jsonSerializer = Mock.Of<IJsonSerializer>();
+        var indexingSettings = Mock.Of<IOptionsMonitor<IndexingSettings>>();
+        Mock.Get(indexingSettings).Setup(x => x.CurrentValue).Returns(new IndexingSettings { });
+        var contentTypeService = Mock.Of<IContentTypeService>();
+        var logger = Mock.Of<ILogger<RichTextPropertyIndexValueFactory>>();
+        string alias = "richText";
+
+        var factory = new RichTextPropertyIndexValueFactory(
+            propertyEditorCollection,
+            jsonSerializer,
+            indexingSettings,
+            contentTypeService,
+            logger);
+
+        // create a mock property with the rich text value
+        var property = Mock.Of<IProperty>(p => p.Alias == alias
+                                               && (string)p.GetValue(It.IsAny<string>(), It.IsAny<string>(),
+                                                   It.IsAny<bool>()) == testContent);
+
+        // get index values
+        var indexValues = factory.GetIndexValues(property, null, null, true, [], new Dictionary<Guid, IContentType>());
+
+        // assert that index values are created correctly
+        var keyValuePairs = indexValues.ToList();
+        Assert.IsNotNull(indexValues);
+        Assert.IsTrue(keyValuePairs.Any());
+        if (!keyValuePairs.Any(kvp => kvp.Key == alias && kvp.Value.FirstOrDefault()?.ToString()?.Contains(expected) == true))
+        {
+            Assert.Fail($"Expected index value for alias '{alias}' to contain '{expected}', but it did not. " +
+                        $"Actual values: {string.Join(", ", keyValuePairs.Select(kvp => $"{kvp.Key}: '{string.Join(", ", kvp.Value)}'"))}");
+        }
```

Comment:
I'd suggest simplifying these assertions a little 😃 

```suggestion
        // get the index value for the property
        var indexValue = factory
            .GetIndexValues(property, null, null, true, [], new Dictionary<Guid, IContentType>())
            .FirstOrDefault(kvp => kvp.Key == alias);
        Assert.IsNotNull(indexValue);

        // assert that index the value is created correctly (it might contain a trailing whitespace, but that's OK)
        var expectedIndexValue = indexValue.Value.SingleOrDefault() as string;
        Assert.IsNotNull(expectedIndexValue);
        Assert.AreEqual(expected, expectedIndexValue.TrimEnd());
```